cleanup: remove Bunch class, refactor some functions, remove deprecated options

dbkeitel opened the pull request:
- some minor help/logging improvements

- remove the various options/methods deprecated in the 1.5.x series.

- more significant cleanup of core module:
     - remove Bunch class which wasn't used except for an isolated test case
     - turn translate_keys_to_lal() into a BaseSearchClass method
     - move read_par() to helper_functions but also add a BaseSearchClass method
     - simplify read_par() by always returning a regular dict

- move predict_fstat() from core to helper_functions

dbkeitel commented:
@GregoryAshton @Rodrigo-Tenorio what do you think?

Rodrigo-Tenorio commented on pyfstat/core.py:
```diff
@@ -371,6 +133,62 @@ def get_output_file_header(self):
+        )
+        return params_dict
+
+    def translate_keys_to_lal(self, dictionary):
```
Should we really bury these "general" functions into a class?
I can take  "Everything relevant in PyFstat is done through `BaseSearchClass`" as a satisfactory answer.

dbkeitel replied:
For this one specifically I didn't really see any use case where someone would need to call it manually, as it's just there to resolve some internal variable name mismatch.

For `read_par()` on the other hand I made both an individually-accesible helper function and the class method which just wraps it, because I can easily imagine using it from an external script to read in pulsar ephemerides and then do something nonstandard with the ranges before calling a Writer or XYZSearch class.

Rodrigo-Tenorio commented on pyfstat/helper_functions.py:
```diff
@@ -628,3 +628,164 @@ def get_parameters_dict_from_file_header(outfile, comments="#", eval_values=Fals
     return params_dict
+
+
+def read_par(
```
But I see you decided to keep this one and not the other one...

Rodrigo-Tenorio replied:
@dbkeitel Are you really keeping this function here just to do a test?
I thought Test classes were derived from BaseSearch so this function can be accessed as an inherited method...?

dbkeitel replied:
See above for why it's a stand-alone function.

The test could indeed be integrated better, however I think we've spent enough time on test cleanups for now. ;)

Rodrigo-Tenorio replied:
No problem: It's a stand-alone function, it has a stand-alone test.

dbkeitel commented:
I'm merging this now for the next release and to unblock the inevitable black version update. @GregoryAshton if you actually still have a good use-case for the `Bunch` class in mind, we can always resurrect it, just let us know.

dbkeitel commented:
Fixed a regression related to parsing of CFSv2 "loudest" files which actually still require `raise_error=False` because they contain complex numbers which `get_dictionary_from_lines` cannot yet parse. Also added a `#FIXME` for that, but the "just ignore errors" workaround should still work for now.

dbkeitel commented:
Tests and an [all-examples run](https://github.com/PyFstat/PyFstat/runs/1052452062?check_suite_focus=true) passed.
